src/Servers/Kestrel/Core/src/Internal/CancellationTokenSourcePool.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    internal class CancellationTokenSourcePool
```

Comment:
Shouldn't this be disposable and dispose all tokens in the queue?

Reply:
I put a note about it in the PR description.

> The implementation doesn't have a dispose. The cost of storing 1024 tokens is tiny and it'll be unrooted once the server is unrooted.